src/scripts/main.js

```diff
@@ -1,3 +1,435 @@
+    document.addEventListener('touchend', this.touchEndHandler);
+
+    this.startButton.addEventListener('click', this.restartHandler);
+    this.startButton.addEventListener('touchend', this.restartHandler);
+  }
+
+  startHandler() {
+    this.startButton.classList.remove('start');
+    this.startButton.classList.add('restart');
+    this.startButton.innerText = 'Restart';
+    this.gameWinMessage.classList.add('hidden');
+    this.gameLoseMessage.classList.add('hidden');
+
+    this.play();
+  }
+
+  restartHandler() {
+    if (this.highScoreValue < this.score) {
+      this.highScoreValue = this.score;
+      this.bestScore.innerText = `${this.highScoreValue}`;
+    }
+
+    this.field = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    for (let j = 0; j < this.field.length; j++) {
+      for (let i = 0; i < this.field[0].length; i++) {
+        this.addNumberOnField(j, i);
+      }
+    }
+
+    this.score = 0;
+
+    this.setScore(this.score);
+
+    this.addNumber();
+    this.addNumber();
+
+    this.gameWinMessage.classList.add('hidden');
+    this.gameLoseMessage.classList.add('hidden');
+  }
+
+  touchStartHandler(e) {
+    e.preventDefault();
+
+    this.startCoords.x = e.screenX;
+    this.startCoords.y = e.screenY;
+
+    if (e.changedTouches) {
+      this.startCoords.x = e.changedTouches[0].screenX;
+      this.startCoords.y = e.changedTouches[0].screenY;
+    }
+
+    return this.startCoords;
+  }
+
+  touchEndHandler(e) {
+    e.preventDefault();
+
+    const copyField = [...this.field].flat();
+
+    let mouseUpX = e.screenX;
+    let mouseUpY = e.screenY;
+
+    if (e.changedTouches) {
+      mouseUpX = e.changedTouches[0].screenX;
+      mouseUpY = e.changedTouches[0].screenY;
+    }
+
+    const deltaX = this.startCoords.x - mouseUpX;
+    const deltaY = this.startCoords.y - mouseUpY;
+
+    if (Math.abs(deltaX) < 30 && Math.abs(deltaY) < 30) {
+      return;
+    }
+
+    if (Math.abs(deltaY) > Math.abs(deltaX)) {
+      if (deltaY > 0) {
+        this.verticalUp();
+      } else {
+        this.verticalDown();
+      }
+    } else {
+      if (deltaX > 0) {
+        this.horizontalLeft();
+      } else {
+        this.horizontalRight();
+      }
+    }
+
+    const fieldAfterMove = [...this.field].flat();
+
+    for (let i = 0; i < copyField.length; i++) {
+      if (copyField[i] !== fieldAfterMove[i]) {
+        this.addNumber();
+        break;
+      }
+    }
+
+    this.setScore(this.score);
+    this.checkOnLose();
+    this.checkOnWin();
+  }
+
+  keyHandler(e) {
+    const copyField = [...this.field].flat();
+
+    if (e.key === 'ArrowUp') {
+      this.verticalUp();
+    } else if (e.key === 'ArrowDown') {
+      this.verticalDown();
+    } else if (e.key === 'ArrowLeft') {
+      this.horizontalLeft();
+    } else if (e.key === 'ArrowRight') {
+      this.horizontalRight();
+    } else {
+      return;
```

Comment:
use switch / case if you need to call some functionality depending on the value of the one variable